src/Type/Definition/Directive.php

```diff
@@ -31,7 +31,7 @@ class Directive
     /** @var string|null */
     public $description;
 
-    /** @var FieldArgument[] */
+    /** @var array<int, FieldArgument> */
```

Comment:
Can we use `list<>`?

Reply:
Perhaps, let's discuss this further in https://github.com/webonyx/graphql-php/issues/811. If we decide to use it, we can make a sweeping change throughout the codebase and replace every instance of `array<int, ` with `list<`.

As far as this PR is concerned, i don't want it blocked by this issue and am thus using the same notation that we already use in a few dozen places.

Reply:
> i don't want it blocked by this issue

yup, np. 

_But we can't just change `array<int, T>` to `list<T>` as it's not the same._